ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Worker.java

```diff
@@ -118,20 +120,33 @@ public void run() {
           singleRun.cancel(true);
           executor.shutdownNow();
           executor = getTimeoutHandlingExecutor();
+          err = true;
         } catch (ExecutionException e) {
           LOG.info("Exception during executing compaction", e);
+          err = true;
         } catch (InterruptedException ie) {
           // do not ignore interruption requests
           return;
```

Comment:
what happens in case of InterruptedException?

Reply:
Since the thread is no longer a daemon thread, in case of InterruptedException (shutdown/restart), the execution needs to be finished.

Reply:
should we at least set the thread as interrupted?

Reply:
should wee at least restore the interrupted status here?

Reply:
At this point the thread will end for sure since we are exiting the main loop. If we restore the interrupted status, InterruptedException will be re-thrown if the thread gets blocked by sleep/wait etc. I'm not sure about it, but the msc.close() call in the finally block may result in an InterruptedException if the threads needs to wait for sth during  closing the Thrift client.

Reply:
Checked and msc.close does not throw any exception, so we can restore the state.

Reply:
Restored the interrupted state